src/notation/internal/masternotation.cpp

```diff
@@ -294,8 +305,7 @@ mu::Ret MasterNotation::setupNewScore(mu::engraving::MasterScore* score, const S
 
     applyOptions(score, scoreOptions);
 
-    m_notationPlayback->init(m_undoStack);
-    initExcerptNotations(score->excerpts());
+    initAfterSettingScore(score);
```

Comment:
This was actually the crucial thing: the following code
```c++
    score->changesChannel().onReceive(this, [this](const ScoreChangesRange&) {
         updateExcerpts();
     });
```
was never called for scores that were newly created by choosing instruments.